### DISCUSSION TITLE
Run command ec2-instance-connect ssh

### DISCUSSION THREAD

**Discussion by sandeza-admin** [aws/aws-cli#8026] [Q&A] [answered]
How could I pass commands to SSH tty after connecting through ec2-instance-connect?

I could ssh in to my instance using the command which is private no public iP

`aws ec2-instance-connect ssh --instance-id <id-example> --connection-type eice`

This is working great after setting up the ec2 end points

I wanted to pass a command, since it is based on OpenSSH, in ssh I do something like this 

`ssh ec2-user@1.1.1.1 'apt-get update'`

Similarly i am looking how to pass Linux commands following the SSH

**Comment by aBurmeseDev** [comment, marked as answer]
Hi @sandeza-admin - thanks for reaching out and sorry for not getting to you sooner. I understand that you were able to connect your EC2 instance using OpenSSH client with AWS CLI [command](https://awscli.amazonaws.com/v2/documentation/api/latest/reference/ec2-instance-connect/ssh.html). Can you elaborate more on what you're trying accomplish as I'm trying to understand when you asked 
>How could I pass commands to SSH tty after connecting through ec2-instance-connect?

Here are some of the resources that I think you might find it helpful:
- [Connect to your Linux instance using the AWS CLI](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/AccessingInstancesLinux.html#SSH-using-EC2-Instance-Connect)
- [Enable SSH connections for your Linux WorkSpaces](https://docs.aws.amazon.com/workspaces/latest/adminguide/connect-to-linux-workspaces-with-ssh.html)

Hope that helps!
John

**Comment by sriram-atadew** [comment]
@aBurmeseDev thanks for the reply I am using AWS CLI command ec2-instance-connect which let me to login EC2 instances without private key (not literally) this is working fine. However after login I wanted to pass a command which needs to be executed. I wanted to pass this along with the login command. Similar to RemoteCommand or ssh -t 'command; bash -l'

**Comment by mentics** [comment]
I had this same question. It's not as straightforward, but it appears that piping to it appears to work.

```
echo "ls -al ; exit # or some command line" | aws ec2-instance-connect ssh --instance-id ${INSTANCE_ID}
```

Note that the login banner will be included in the output. If you need the output for something, you might want to skip that using sed or something.
The exit is in there because it will leave you in the shell otherwise.
